automation/gradle/versionCode.gradle

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import java.text.SimpleDateFormat
+
+// This gradle scripts generates a "unique" version code for our release versions.
```

Comment:
@pocmo @csadilek Are you guys fine reusing the same versionCode scheme on reference-browser? Or would you like make any adjustments?

Reply:
Ahh, good point Johan, I shouldn't be declaring the version scheme for `reference-browser`, since it's a long-term decision.
I've created a ticket for the `reference-browser` lads [here](https://github.com/mozilla-mobile/reference-browser/issues/334). As discussed on Vidyo earlier today, we'll merge releng pipeline changes, but we'll leave versioning schemes for now :smile:

Reply:
I think this is fine. I'm open to other proposals though. Our requirements are:

* It should generate a small number since we can only go forward and never backward. That's why we are using that funky date format in Focus, just so that it's a bit shorter.
* An updated version needs to always have a higher version code than the previous release.
* We need some wiggle room for publishing multiple APK builds (Currently x86 and ARM, soon x86_64 and aarch64 too, in the long run we may only need to support one multiarch build): Those builds need different version codes and they need to be in a specific order: x86_64 > x86 > aarch64 > arm. Those different version codes should be close to each other (e.g. no prefix) so that they can be replaced with combined/shared builds later. Currently we just add 1/2/3 to the version code.

The date based version code solves this problem nicely. Unfortunately it makes the builds not 100% reproducible. A rebuild will always have a new version code. Back for Focus I was also looking at options based on git (e.g. count of commits) but that ties it directly to the repository layout and makes it hard to have the architecture gaps as well.

Reply:
FWIW, I've taken care of the "wiggle room" [here, in `build.gradle`](https://github.com/mozilla-mobile/reference-browser/pull/268/files#diff-39e7d8c00954e920b98e7636f0ac30b2R100), as influenced by focus :smile:

Reply:
👍